zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/tenant/TenantUpdateProcessor.java

```diff
@@ -0,0 +1,151 @@
+              .formatted(record.getTenantId()));
+      return;
+    }
+
+    updateExistingTenant(persistedRecord.get(), record);
+    updateStateAndDistribute(command, persistedRecord.get());
+  }
+
+  @Override
+  public void processDistributedCommand(final TypedRecord<TenantRecord> command) {
+    stateWriter.appendFollowUpEvent(
+        command.getValue().getTenantKey(), TenantIntent.UPDATED, command.getValue());
+    commandDistributionBehavior.acknowledgeCommand(command);
+  }
+
+  private boolean isAuthorizedToUpdate(final TypedRecord<TenantRecord> command) {
+    final var authorizationRequest =
+        new AuthorizationRequest(command, AuthorizationResourceType.TENANT, PermissionType.UPDATE)
+            .addResourceId(command.getValue().getTenantId());
+    if (!authCheckBehavior.isAuthorized(authorizationRequest)) {
+      rejectCommandWithUnauthorizedError(command, authorizationRequest);
+      return false;
+    }
+    return true;
+  }
+
+  private boolean tenantIdConflict(final TenantRecord record) {
+    final Optional<Long> tenantKeyById = tenantState.getTenantKeyById(record.getTenantId());
+    return tenantKeyById.isPresent() && !tenantKeyById.get().equals(record.getTenantKey());
+  }
+
+  private void updateExistingTenant(
+      final TenantRecord existingTenant, final TenantRecord updateRecord) {
+    if (!updateRecord.getTenantId().isEmpty()) {
+      existingTenant.setTenantId(updateRecord.getTenantId());
+    }
+    if (!updateRecord.getName().isEmpty()) {
+      existingTenant.setName(updateRecord.getName());
+    }
+  }
+
+  private void updateStateAndDistribute(
+      final TypedRecord<TenantRecord> command, final TenantRecord tenantRecord) {
+    stateWriter.appendFollowUpEvent(
+        tenantRecord.getTenantKey(), TenantIntent.UPDATED, tenantRecord);
+    responseWriter.writeEventOnCommand(
+        tenantRecord.getTenantKey(), TenantIntent.UPDATED, tenantRecord, command);
+    distributeCommand(command);
+  }
+
+  private void distributeCommand(final TypedRecord<TenantRecord> command) {
+    final long distributionKey = keyGenerator.nextKey();
+    commandDistributionBehavior
+        .withKey(distributionKey)
+        .inQueue(DistributionQueue.IDENTITY.getQueueId())
+        .distribute(command);
+  }
+
+  private void rejectCommandWithUnauthorizedError(
+      final TypedRecord<TenantRecord> command, final AuthorizationRequest authorizationRequest) {
+    final var errorMessage =
+        AuthorizationCheckBehavior.UNAUTHORIZED_ERROR_MESSAGE.formatted(
+            authorizationRequest.getPermissionType(), authorizationRequest.getResourceType());
+    rejectCommand(command, RejectionType.UNAUTHORIZED, errorMessage);
+  }
+
+  private void rejectCommand(
+      final TypedRecord<TenantRecord> command,
+      final RejectionType type,
+      final String errorMessage) {
+    rejectionWriter.appendRejection(command, type, errorMessage);
+    responseWriter.writeRejectionOnCommand(command, type, errorMessage);
```

Comment:
👍 I like the nice helper method. I noticed that we usually duplicated these two lines in our processors.